### ISSUE TITLE
Malformed aws_security_group_rule causes crash, not syntax error - tf 0.6.4

### ISSUE THREAD

**Issue by packplusplus** [hashicorp/terraform#3577]
If you use some malformed syntax in a security_group rule, you don't get syntax error, you get a crash. 

```  
provider "aws" {
}

resource "aws_security_group" "sg_test" {
  name = "sg_test"
}

resource "aws_security_group_rule" "sg_test_rule" {
  type = "ingress"
  from_port = 22
  to_port = 22
  protocol = "tcp"
  security_group_id = [ "${aws_security_group.sg_test.id}" ]
}
``` 

You can use this example to get the crash log, but the main error appears to be:
```
panic: Error reading level config: '' expected type 'string', got unconvertible type '[]interface {}'
```

```
terraform --version
Terraform v0.6.4
```

**Comment by catsby** [comment]
Thanks @packplusplus – this is related to validating computed values, in your example `security_group_id` is both malformed and computed. If you were to manually hardcode `security_group_id` you'd get the regular syntax error. 

Fortunately, I just confirmed that this issue is fixed in #3009, so we'll close this when that gets merged.

Thanks for opening the issue!

**Comment by packplusplus** [comment]
+1 thanks @catsby

**Comment by apparentlymart** [comment]
Fixed by 8fdffb65eab30162061af4bb85fbf5ba6db9bf6e

**Comment by catsby** [comment]
Thanks for the follow up, @apparentlymart

**Comment by packplusplus** [comment]
Verified fixed in 0.6.6. You're both great! Crashes scare me.
